tests/emukit/bayesian_optimization/test_local_penalization.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from scipy.optimize import check_grad
+
+from emukit.bayesian_optimization.acquisitions.local_penalization import LocalPenalization
+from emukit.core.interfaces import IModel
+
+TOL = 1e-6
+
+
+def test_penalization_function():
+
+    model = MockModel()
+    lp = LocalPenalization(model)
+    lp.update_batches(np.zeros((5, 1)), 1, -0.1)
+
+    value = lp.evaluate(np.random.rand(10, 1))
+    assert value.shape == (10, 1)
+
+
+def test_penalization_function_gradients_shape():
+
+    model = MockModel()
+    lp = LocalPenalization(model)
+    lp.update_batches(np.zeros((5, 2)), 1, -0.1)
+
+    val, grad = lp.evaluate_with_gradients(np.random.rand(10, 2))
+    assert grad.shape == (10, 2)
+    assert val.shape == (10, 1)
+
+
+def test_local_penalization_gradients_with_single_point_in_batch():
+    np.random.seed(123)
+    model = MockModel()
+    lp = LocalPenalization(model)
+    lp.update_batches(np.zeros((1, 1)), 1, -0.1)
+
+    x0 = np.array([0.5])
+    _check_grad(lp, TOL, x0)
+
+
+def test_local_penalization_gradients_with_no_batch():
+    np.random.seed(123)
+    model = MockModel()
+    lp = LocalPenalization(model)
+    lp.update_batches(np.zeros((1, 1)), 1, -0.1)
+
+    x0 = np.array([0.5])
+    _check_grad(lp, TOL, x0)
+
+
+def test_local_penalization_gradients_with_multiple_batch():
```

Comment:
I do not understand the function name of this one nor the one above. Is it a numerical gradient check for a single datapoint in batch and  for multiple datapoints in the batch?